Node 16 upgrade of ts-transformer breaks due to strict peerDependency

<!-- If this is more of a question on how i18n should work or you have not narrowed down the issue to be 1 of our packages, please use Discussions instead -->

## Which package?
Since this is a monorepo. Please specify which package this bug report is for.

ts-transformer

## Describe the bug
The package has a peer dependency on `ts-jest@27` and my repo is using `ts-jest@28` which, when I do an upgrade to the latest version using node 16, it blows up as follows:
```
npm ERR! While resolving: @formatjs/ts-transformer@3.9.7
npm ERR! Found: ts-jest@28.0.3
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"28.0.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional ts-jest@"27" from @formatjs/ts-transformer@3.9.7
npm ERR! node_modules/@formatjs/ts-transformer
npm ERR!   dev @formatjs/ts-transformer@"3.9.7" from the root project
npm ERR!   @formatjs/ts-transformer@"3.9.7" from babel-plugin-formatjs@10.3.23
npm ERR!   node_modules/babel-plugin-formatjs
npm ERR!     dev babel-plugin-formatjs@"10.3.23" from the root project
npm ERR!   1 more (eslint-plugin-formatjs)
npm ERR! 
npm ERR! Conflicting peer dependency: ts-jest@27.1.5
npm ERR! node_modules/ts-jest
npm ERR!   peerOptional ts-jest@"27" from @formatjs/ts-transformer@3.9.7
npm ERR!   node_modules/@formatjs/ts-transformer
npm ERR!     dev @formatjs/ts-transformer@"3.9.7" from the root project
npm ERR!     @formatjs/ts-transformer@"3.9.7" from babel-plugin-formatjs@10.3.23
npm ERR!     node_modules/babel-plugin-formatjs
npm ERR!       dev babel-plugin-formatjs@"10.3.23" from the root project
npm ERR!     1 more (eslint-plugin-formatjs)
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

## Expected behavior
Consider changing the `peerDependencies` for `ts-jest` in the `ts-transformer` package to be `>= 27` or eliminate it if it really isn't needed

### NOTE
I WAS able to work-around the issue by using `--legacy-peer-deps` but locking something like this down to one version of a common testing library seems problematic for future upgrades for people using `node 16`

--- a/packages/ts-transformer/package.json
+++ b/packages/ts-transformer/package.json
@@ -26,7 +26,7 @@
     "typescript": "^4.5"
   },
   "peerDependencies": {
-    "ts-jest": "27"
+    "ts-jest": "27 || 28"
   },
   "peerDependenciesMeta": {
     "ts-jest": {

